### PR TITLE
Pn 3866 fe gestire gruppi in fase di invio manuale della notifica

### PR DESCRIPTION
**Pull request by NicolaGiornetta**
## Short description
Added hasGroups parameter to auth.api of PA and modified validation of Preliminary Informations

## List of changes proposed in this pull request
Accordi to the task, the preliminary informations have now new validation for groups. If hasGroups in userData is present and true, the field is set as required, even if there are groups available.

## How to test
Two tests:
- You have to find a user who have hasGroups set to true and then preliminary informations will required the field to continue
- You have to find a user who have hasGroups set to false and then preliminary informations will NOT require the field to continue

### CODE REVIEW

File: packages/pn-pa-webapp/src/pages/components/NewNotification/PreliminaryInformations.tsx [hunk 1]
```diff
@@ -54,7 +55,7 @@ const PreliminaryInformations = ({ notification, onConfirm }: Props) => {
     subject: yup.string().required(`${t('subject')} ${tc('required')}`),
     physicalCommunicationType: yup.string().required(),
     paymentMode: yup.string().required(),
-    group: groups.length > 0 ? yup.string().required() : yup.string(),
+    group: (groups.length > 0 && user.hasGroups) ? yup.string().required() : yup.string(),
```
Comment (LorenzoCocciarelli) [inline packages/pn-pa-webapp/src/pages/components/NewNotification/PreliminaryInformations.tsx:58]: we could avoid checking for groups length using hasGroups, if hasGroups is false groups.length would alsa be 0

File: packages/pn-pa-webapp/src/api/auth/Auth.api.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export const AuthApi = {
             fiscal_number: response.data.fiscal_number,
             organization: response.data.organization,
             sessionToken: response.data.sessionToken,
-            uid: response.data.uid
+            uid: response.data.uid,
+            hasGroups: response.data.hasGroups,
```
Comment (LorenzoCocciarelli) [inline packages/pn-pa-webapp/src/api/auth/Auth.api.ts:17]: hasGroups should be a field inside organization

File: packages/pn-pa-webapp/src/redux/auth/reducers.ts [hunk 3]
```diff
@@ -29,6 +29,7 @@ const organizationMatcher = yup.object({
 const userDataMatcher = yup
   .object({
     ...basicUserDataMatcherContents,
+    hasGroups: yup.boolean(),
```
Comment (LorenzoCocciarelli) [inline packages/pn-pa-webapp/src/redux/auth/reducers.ts:32]: hasGroups should be inside the organizationMatcher

**Review by LorenzoCocciarelli — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/api/auth/Auth.api.ts (hunk 2 above)
Comment (NicolaGiornetta) [reply packages/pn-pa-webapp/src/api/auth/Auth.api.ts:17]: so, I should have response.date.organization.hasGroup instead?

Comment (LorenzoCocciarelli) [reply packages/pn-pa-webapp/src/api/auth/Auth.api.ts:17]: yes, it should

**Review by LorenzoCocciarelli — APPROVED** [review]

**Pull request merged**